File: packages/python/examples/pytest/locator_test.py
def test_locator(al, navigate, type):
    navigate("https://bonigarcia.dev/selenium-webdriver-java/web-form.html")

    text_input = al.find("text input")
    assert text_input is not None
    type(text_input, "Hello Alumnium!")

    textarea = al.find("textarea")
    assert textarea is not None
    type(textarea, "Testing the LocatorAgent")

    submit_button = al.find("submit button")
    assert submit_button is not None
    submit_button.click()

File: packages/java/src/test/java/ai/alumnium/system/LocatorTest.java
package ai.alumnium.system;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

public class LocatorTest extends BaseTest {

  private static final String FORM_URL =
      "https://bonigarcia.dev/selenium-webdriver-java/web-form.html";

  @Test
  void testFindsElements() {
    navigate(FORM_URL);

    Object textInput = al.find("text input");
    Assertions.assertNotNull(textInput);
    type(textInput, "Hello Alumnium!");

    Object textarea = al.find("textarea");
    Assertions.assertNotNull(textarea);
    type(textarea, "Testing the LocatorAgent");

    Object submitButton = al.find("submit button");
    Assertions.assertNotNull(submitButton);
    click(submitButton);
  }
}
